### DISCUSSION TITLE
Github Actions fails to deploy my github website

### DISCUSSION THREAD

**Discussion by scvcrem** [community/community#101754] [Actions]
### Select Topic Area

Question

### Body

I have commited an update to my github website. Github Actions gives me the following error when trying to deploy the new version. Even reverting all the new commits, deployment still does not work. 

Run yes | bash bin/deploy --verbose  --src master --deploy gh-pages
+ shift
+ [[ 4 > 0 ]]
Deploying...
+ key=--src
+ case $key in
+ SRC_BRANCH=master
+ shift
+ shift
+ [[ 2 > 0 ]]
+ key=--deploy
+ case $key in
+ DEPLOY_BRANCH=gh-pages
+ shift
+ shift
+ [[ 0 > 0 ]]
+ set -e
+ echo Deploying...
+ echo 'Source branch: master'
+ echo 'Deploy branch: gh-pages'
+ read -r -p 'Do you want to proceed? [y/N] ' response
Source branch: master
Deploy branch: gh-pages
+ [[ ! y =~ ^([yY][eE][sS]|[yY])+$ ]]
+ git diff-index --quiet HEAD --
++ git ls-files --exclude-standard --others
+ test -z ''
++ git branch
++ grep master
++ tr ' ' '\n'
++ tail -1
+ '[' master ']'
+ git checkout master
Already on 'master'
Your branch is up to date with 'origin/master'.
++ git branch
++ grep gh-pages
+ '[' ']'
+ git checkout -b gh-pages
Switched to a new branch 'gh-pages'
+ bundle exec jekyll build
bundler: failed to load command: jekyll (/home/runner/work/scvcrem.github.io/scvcrem.github.io/vendor/bundle/ruby/3.0.0/bin/jekyll)
/opt/hostedtoolcache/Ruby/3.0.2/x64/lib/ruby/gems/3.0.0/gems/bundler-2.5.5/lib/bundler/runtime.rb:304:in `check_for_activated_spec!': You have already activated uri 0.10.1, but your Gemfile requires uri 0.13.0. Since uri is a default gem, you can either remove your dependency on it or try updating to a newer version of bundler that supports uri as a default gem. (Gem::LoadError)
	from /opt/hostedtoolcache/Ruby/3.0.2/x64/lib/ruby/gems/3.0.0/gems/bundler-2.5.5/lib/bundler/runtime.rb:25:in `block in setup'
	from /opt/hostedtoolcache/Ruby/3.0.2/x64/lib/ruby/gems/3.0.0/gems/bundler-2.5.5/lib/bundler/spec_set.rb:191:in `each'
	from /opt/hostedtoolcache/Ruby/3.0.2/x64/lib/ruby/gems/3.0.0/gems/bundler-2.5.5/lib/bundler/spec_set.rb:191:in `each'
	from /opt/hostedtoolcache/Ruby/3.0.2/x64/lib/ruby/gems/3.0.0/gems/bundler-2.5.5/lib/bundler/runtime.rb:24:in `map'
	from /opt/hostedtoolcache/Ruby/3.0.2/x64/lib/ruby/gems/3.0.0/gems/bundler-2.5.5/lib/bundler/runtime.rb:24:in `setup'
	from /opt/hostedtoolcache/Ruby/3.0.2/x64/lib/ruby/gems/3.0.0/gems/bundler-2.5.5/lib/bundler.rb:162:in `setup'
	from /opt/hostedtoolcache/Ruby/3.0.2/x64/lib/ruby/gems/3.0.0/gems/bundler-2.5.5/lib/bundler/setup.rb:26:in `block in <top (required)>'
	from /opt/hostedtoolcache/Ruby/3.0.2/x64/lib/ruby/gems/3.0.0/gems/bundler-2.5.5/lib/bundler/ui/shell.rb:159:in `with_level'
	from /opt/hostedtoolcache/Ruby/3.0.2/x64/lib/ruby/gems/3.0.0/gems/bundler-2.5.5/lib/bundler/ui/shell.rb:111:in `silence'
	from /opt/hostedtoolcache/Ruby/3.0.2/x64/lib/ruby/gems/3.0.0/gems/bundler-2.5.5/lib/bundler/setup.rb:26:in `<top (required)>'
	from /opt/hostedtoolcache/Ruby/3.0.2/x64/lib/ruby/gems/3.0.0/gems/bundler-2.5.5/lib/bundler/cli/exec.rb:56:in `require_relative'
	from /opt/hostedtoolcache/Ruby/3.0.2/x64/lib/ruby/gems/3.0.0/gems/bundler-2.5.5/lib/bundler/cli/exec.rb:56:in `kernel_load'
	from /opt/hostedtoolcache/Ruby/3.0.2/x64/lib/ruby/gems/3.0.0/gems/bundler-2.5.5/lib/bundler/cli/exec.rb:23:in `run'
	from /opt/hostedtoolcache/Ruby/3.0.2/x64/lib/ruby/gems/3.0.0/gems/bundler-2.5.5/lib/bundler/cli.rb:451:in `exec'
	from /opt/hostedtoolcache/Ruby/3.0.2/x64/lib/ruby/gems/3.0.0/gems/bundler-2.5.5/lib/bundler/vendor/thor/lib/thor/command.rb:28:in `run'
	from /opt/hostedtoolcache/Ruby/3.0.2/x64/lib/ruby/gems/3.0.0/gems/bundler-2.5.5/lib/bundler/vendor/thor/lib/thor/invocation.rb:127:in `invoke_command'
	from /opt/hostedtoolcache/Ruby/3.0.2/x64/lib/ruby/gems/3.0.0/gems/bundler-2.5.5/lib/bundler/vendor/thor/lib/thor.rb:527:in `dispatch'
	from /opt/hostedtoolcache/Ruby/3.0.2/x64/lib/ruby/gems/3.0.0/gems/bundler-2.5.5/lib/bundler/cli.rb:34:in `dispatch'
	from /opt/hostedtoolcache/Ruby/3.0.2/x64/lib/ruby/gems/3.0.0/gems/bundler-2.5.5/lib/bundler/vendor/thor/lib/thor/base.rb:584:in `start'
	from /opt/hostedtoolcache/Ruby/3.0.2/x64/lib/ruby/gems/3.0.0/gems/bundler-2.5.5/lib/bundler/cli.rb:28:in `start'
	from /opt/hostedtoolcache/Ruby/3.0.2/x64/lib/ruby/gems/3.0.0/gems/bundler-2.5.5/exe/bundle:28:in `block in <top (required)>'
	from /opt/hostedtoolcache/Ruby/3.0.2/x64/lib/ruby/gems/3.0.0/gems/bundler-2.5.5/lib/bundler/friendly_errors.rb:117:in `with_friendly_errors'
	from /opt/hostedtoolcache/Ruby/3.0.2/x64/lib/ruby/gems/3.0.0/gems/bundler-2.5.5/exe/bundle:20:in `<top (required)>'
	from /opt/hostedtoolcache/Ruby/3.0.2/x64/bin/bundle:23:in `load'
	from /opt/hostedtoolcache/Ruby/3.0.2/x64/bin/bundle:23:in `<main>'
yes: standard output: Broken pipe
Error: Process completed with exit code 1.

**Comment by AliAlmasi** [comment]
[This may have an answer for you](https://stackoverflow.com/questions/69890412/bundler-failed-to-load-command-jekyll).

**Comment by muizzhaque** [comment]
To resolve this issue, you can try the following steps:

1..Update your Gemfile.lock file to ensure that it reflects the correct versions of gems required by your project.

2.Review your Gemfile and Gemfile.lock to identify any conflicting versions of the uri gem. You may need to specify a specific version of the uri gem in your Gemfile to resolve the conflict.

3.Run bundle update uri to update the uri gem to the version specified in your Gemfile.

4.After updating the Gemfile or Gemfile.lock, run bundle install to install the updated gems.

5.Double-check the Gemfile.lock after running bundle update and bundle install to ensure that the correct versions of gems are installed.

6.If the issue persists, try clearing the Bundler cache by running bundle clean --force and then re-run bundle install.

7.Review other dependencies in your project to ensure there are no other conflicting gem versions.

8.After resolving the gem version conflict, retry the deployment process to see if the issue is resolved.

Hope this steps might be helpful
